irony-cdb-json.el

```diff
@@ -77,28 +77,55 @@ directories to project directory."
     (let ((elm (nth pos target-list)))
       (append (list elm) (delete elm target-list)))))
 
+(defun irony-cdb-json--cdb-list ()
+  (mapcar (lambda (x) (cdr x)) irony-cdb-json--project-alist))
+
 (defun irony-cdb-json--choose-cdb ()
-  (let ((choices (mapcar (lambda (x) (cdr x)) irony-cdb-json--project-alist)))
-    (completing-read "Choose Irony CDB: " choices nil 'require-match nil)))
+  (let ((cdbs (irony-cdb-json--cdb-list)))
+    (if (> (length cdbs) 0)
+        (completing-read "Choose Irony CDB: " cdbs nil 'require-match nil)
+      nil)))
 
 ;;;###autoload
-(defun irony-cdb-json-select ()
+(defun irony-cdb-json-select (&optional cdb)
   "Select CDB to use with a prompt.
 
 It is useful when you have several CDBs with the same project
 root.
 
 The completion function used internally is `completing-read' so
 it could easily be used with helm, for instance, by enabling
-`helm-mode' before calling the function."
+`helm-mode' before calling the function using an advice."
```

Comment:
It is because helm mode isn't turned on unless you're doing completions and therefore I only want to turn it on if it isn't already. But I'll remove that comment if it is confusing.